src/nni_manager/training_service/local/gpuScheduler.ts

```diff
@@ -100,12 +100,20 @@ class GPUScheduler {
     }
 
     private async updateGPUSummary(): Promise<void> {
-        const cmdresult: cpp.childProcessPromise.Result =
-            await execTail(path.join(this.gpuMetricCollectorScriptFolder, 'gpu_metrics'));
-        if (cmdresult && cmdresult.stdout) {
-            this.gpuSummary = <GPUSummary>JSON.parse(cmdresult.stdout);
-        } else {
-            this.log.error('Could not get gpu metrics information!');
+        try {
+            let gpuMetricPath = path.join(this.gpuMetricCollectorScriptFolder, 'gpu_metrics');
+            if (fs.existsSync(gpuMetricPath)) {
+                const cmdresult: cpp.childProcessPromise.Result = await execTail(gpuMetricPath);
+                if (cmdresult && cmdresult.stdout) {
+                    this.gpuSummary = <GPUSummary>JSON.parse(cmdresult.stdout);
+                } else {
+                    this.log.error('Could not get gpu metrics information!');
+                }
+            } else{
+                this.log.warning('gpu_metrics file does not exist!')
+            }
+        } catch(err) {
+            this.log.error(err);
```

Comment:
Please remove this new outside try catch if it is unnecessary.

Reply:
removed.